server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongKeyedBucketOrds.java

```diff
@@ -49,6 +52,8 @@ public static LongKeyedBucketOrds buildForValueRange(BigArrays bigArrays, Cardin
         });
     }
 
+    private TreeSet<Long> keySet = null;
```

Comment:
Once this is set you can't add any more things to it. It's probably worth asserting that.

I wonder if this is better as a `long[]` and then you sort it. That feels lighter than all of the tree stuff.

Reply:
I thought about a `long[]`, but we'd still need duplicate removal.  Which is doable, but seemed like enough complexity to justify the tree set.  Open to trying it that way though.

Reply:
To be clear, we need duplicate removal *at insert time*, otherwise it ends up being just as big as the `ords` array, basically by definition

Reply:
Maybe it would be good to nullify this variable during release?